src/main/java/org/springframework/data/domain/ExampleSpec.java

```diff
@@ -0,0 +1,797 @@
+	 *
+	 * @author Mark Paluch
+	 */
+	public static class GenericPropertyMatchers {
+
+		/**
+		 * Creates a {@link GenericPropertyMatcher} that matches string case insensitive.
+		 *
+		 * @return
+		 */
+		public static GenericPropertyMatcher ignoreCase() {
+			return new GenericPropertyMatcher().ignoreCase();
+		}
+
+		/**
+		 * Creates a {@link GenericPropertyMatcher} that matches string case sensitive.
+		 *
+		 * @return
+		 */
+		public static GenericPropertyMatcher caseSensitive() {
+			return new GenericPropertyMatcher().caseSensitive();
+		}
+
+		/**
+		 * Creates a {@link GenericPropertyMatcher} that matches string using {@link StringMatcher#CONTAINING}.
+		 *
+		 * @return
+		 */
+		public static GenericPropertyMatcher contains() {
+			return new GenericPropertyMatcher().contains();
+		}
+
+		/**
+		 * Creates a {@link GenericPropertyMatcher} that matches string using {@link StringMatcher#ENDING}.
+		 *
+		 * @return
+		 */
+		public static GenericPropertyMatcher endsWith() {
+			return new GenericPropertyMatcher().endsWith();
+
+		}
+
+		/**
+		 * Creates a {@link GenericPropertyMatcher} that matches string using {@link StringMatcher#STARTING}.
+		 *
+		 * @return
+		 */
+		public static GenericPropertyMatcher startsWith() {
+			return new GenericPropertyMatcher().startsWith();
+		}
+
+		/**
+		 * Creates a {@link GenericPropertyMatcher} that matches string using {@link StringMatcher#EXACT}.
+		 *
+		 * @return
+		 */
+		public static GenericPropertyMatcher exact() {
+			return new GenericPropertyMatcher().startsWith();
+		}
+
+		/**
+		 * Creates a {@link GenericPropertyMatcher} that matches string using {@link StringMatcher#DEFAULT}.
+		 *
+		 * @return
+		 */
+		public static GenericPropertyMatcher storeDefaultMatching() {
+			return new GenericPropertyMatcher().storeDefaultMatching();
+		}
+
+		/**
+		 * Creates a {@link GenericPropertyMatcher} that matches string using {@link StringMatcher#REGEX}.
+		 *
+		 * @return
+		 */
+		public static GenericPropertyMatcher regex() {
+			return new GenericPropertyMatcher().regex();
+		}
+
+	}
+
+	/**
+	 * Match modes for treatment of {@link String} values.
+	 *
+	 * @author Christoph Strobl
+	 */
+	public static enum StringMatcher {
+
+		/**
+		 * Store specific default.
+		 */
+		DEFAULT(null),
+		/**
+		 * Matches the exact string
+		 */
+		EXACT(Type.SIMPLE_PROPERTY),
+		/**
+		 * Matches string starting with pattern
+		 */
+		STARTING(Type.STARTING_WITH),
+		/**
+		 * Matches string ending with pattern
+		 */
+		ENDING(Type.ENDING_WITH),
+		/**
+		 * Matches string containing pattern
+		 */
+		CONTAINING(Type.CONTAINING),
+		/**
+		 * Treats strings as regular expression patterns
+		 */
+		REGEX(Type.REGEX);
+
+		private Type type;
+
+		private StringMatcher(Type type) {
+			this.type = type;
+		}
+
+		/**
+		 * Get the according {@link Part.Type}.
+		 *
+		 * @return {@literal null} for {@link StringMatcher#DEFAULT}.
+		 */
+		public Type getPartType() {
```

Comment:
We can't refer to the `query.parser` package here as it's a less fundamental package and would introduce a cycle. I couldn't really find usage of the method in Spring Data Commons itself. Maybe we can invert the dependency by introducing a translating mechanism in the `query.parser` package?